mass/core/query_handler.py

```diff
@@ -28,18 +33,87 @@ def __init__(
         self,
         *,
         aggregator_collection: AggregatorCollectionPort,
-        dao_collection: DaoCollectionPort
+        dao_collection: DaoCollectionPort,
     ):
         """Initialize the query handler with resource daos/aggregators"""
         self._aggregator_collection = aggregator_collection
         self._dao_collection = dao_collection
 
+    def _get_nested_fields(self, *, fields: List) -> Set:
```

Comment:
Please provide detailed type annotations:
```suggestion
    def _get_nested_fields(self, *, fields: List[str]) -> Set[tuple[str,str]]:
```